### PR TITLE
prefix() proposal.  Closes broadinstitute/cromwell#375

### PR DESCRIPTION
**Pull request by mcovarr**
This is an attempt to solve the problem presented [here](https://github.com/broadinstitute/cromwell/issues/375) of composing a string from a prefix string and a list of zero or more value strings.  There are definitely a lot of ways this problem might be solved and this is a concrete proposal just to get a discussion started.  😄 

Pinging @slnovak @vdauwera @knoblett @katevoss @jsotobroad @LeeTL1220 @yfarjoun

### CODE REVIEW

**Comment by Horneth** [comment]
I also like the idea of having an `mkString` for arrays in general. I find the scala one pretty simple and complete but it doesn't have to be the same. 
http://www.scala-lang.org/api/2.10.3/#scala.collection.GenTraversableOnce

And we could still have `prefix`, `suffix` sugar method on top.

File: SPEC.md [hunk 1]
```diff
@@ -94,6 +94,7 @@
   * [Array\[Array\[X\]\] transpose(Array\[Array\[X\]\])](#arrayarrayx-transposearrayarrayx)
   * [Pair(X,Y) zip(X,Y)](#pairxy-zipxy)
   * [Pair(X,Y) cross(X,Y)](#pairxy-crossxy)
+  * [String prefix(String, Array\[String\])](#string-prefix-string-arraystring)
```
Comment (cjllanwarne) [inline SPEC.md]: This link didn't work for me

File: SPEC.md [hunk 2]
```diff
@@ -2491,6 +2492,18 @@ Array[String] zs = [ "d", "e" ]
...
+## String prefix(String, Array[String])
+
+:pig2: [Cromwell supported](https://github.com/broadinstitute/cromwell#wdl-support) :white_check_mark:
+
+Given a String and an Array[String], the `prefix` function returns a String comprised of each element of the 
+Array prefixed by the String, all joined together.  For example:
+   
+```
+Array[String] env = ["key1=value1", "key2=value2", "key3=value3"]
+String env_param = prefix(" -e ", env) # " -e key1=value1 -e key2=value2 -e key3=value3"
```
Comment (cjllanwarne) [inline SPEC.md]: Nice! The extra space at the start is awkward but (shrug)

File: SPEC.md (hunk 1 above)
Comment (mcovarr) [inline SPEC.md]: Whoops, that's because I have a superflous dash.  Thank you for reviewing so thoroughly!  😄

**Comment by mcovarr** [comment]
@cjllanwarne a Scala-esque `mkString` wouldn't make this redundant since `mkString` only inserts strings between elements of the array, it doesn't prepend the strings to every element of the array.  For this particular use case, I want a completely blank string if the array is empty, and ` -e key1=value` if there's one element in the array.

Having said that I'm happy to make `mkString`, `prefix`, and `suffix` functions as @Horneth suggested if we're agreed those would be a good idea.

**Comment by cjllanwarne** [comment]
@mcovarr good point! I think `prefix` adds by far the most value. `mkString` is nice (especially if it means we can retire the `${sep=" " array}` syntax although I would prefer a better name. Suffix, meh, but there might be uses for it so why not...

**Comment by Horneth** [comment]
👍

**Comment by cjllanwarne** [comment]
👍

**Pull request merged**